pyramus/src/main/java/fi/otavanopisto/pyramus/json/students/EditStudentJSONRequestController.java

```diff
@@ -523,7 +532,38 @@ else if (phoneId > 0 && number != null) {
           ApplicationUtils.deleteApplication(application);
         }
       }
+      
+      // Student card
+      
+      StudentCard studentCard = studentCardDAO.findByStudent(student);
+
+      Date expiryDate = null;
+      
+      // Set expiry date automatically same as study end date or study time end
+      if (student.getStudyEndDate() != null) {
+        expiryDate = student.getStudyEndDate();
+      }  else {
+        expiryDate = student.getStudyTimeEnd();
+      }
+      
+      if (studentCard != null && studentCardType != null) {
```

Comment:
Could the latter condition be moved up? Looks like `studentCardType` would always need to exist but if it doesn't, it means nothing student card related will happen, anyway.